### ISSUE TITLE
feat[dev]: ✨ implement A11y improvements to sd-step-group and sd-step

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#1487]
## Description
This task describes the Storybook implementation for the sd-step-group and sd-step component as part of the A11y improvements outlined in Epic https://github.com/solid-design-system/solid/issues/1465.

A11y improvements for sd-step-group and sd-step can be found here: [sd-sd-step-group and sd-step Notes](https://www.notion.so/matuzo/sd-step-group-sd-step-10df3f6e1bc080c68d97eb08e7c3f1af).

----------------------------------

## Comment
**Problem:** There shouldn't be a mismatch between the information I get depending on my chosen navigation type. The current step is not focusable via keyboard (`tabindex=-1`) but accessible via virtual cursor.
**Solution:** Don't use a button for non-interactive steps (or remove `tabindex`).

**Problem:** Screen reader users can't obtain information about future steps because future steps are disabled.
**Solution:**  Don't use a button for non-interactive steps (or remove disabled).

**Problem:** Screen reader users cannot determine how many steps there are.
**Solution:** Get rid of disabled buttons (see previous problems) and wrap steps in `<ol/li>`. Lists provide screen reader users natively with the number of list items.
Remember to set the roles correct on step-group (list) and step (item).

**Problem:** Horizontal step groups are not responsive, which causes overflow on small viewports or when the page is zoomed in.
**Solution:** its the users responisbility => comment in the description should be added (tba by @coraliefeil)

> have a look into #1154 in order may be combine the fixes.

## Notes
- Just to make sure that you're aware: The label and description slots don't influence the accessible name of the button. It's always "Step x".
- There is no connection between the button and, the label and description. Consider connecting them using aria-describedby.
```
<button
	part="circle"
	tabindex="0"
	aria-label="Step 1"
	aria-describedby="text-container"
></button>

<div part="text-container" id="text-container">
  <div part="label">
		<slot name="label"></slot>
	</div>
	<div part="description">
		<slot></slot>
	</div>
</div>
```

- It's a good practice to provide a label for navigation landmarks. It would be best if it could be defined via an attribute.

## Template Ticket:

- [ ] Pick up the template ticket when working on the component story for A11y improvement, as it references the component and will be resolved once the component is made accessible:
https://github.com/solid-design-system/solid/issues/1523

## Additional ressources
- https://css-tricks.com/making-disabled-buttons-more-inclusive/
- https://axesslab.com/disabled-buttons-suck/

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

**Comment by MartaPintoTeixeira** [comment]
While doing this task please take care of the template as well before closing ticket.

**Comment by MartaPintoTeixeira** [comment]
"Hint: Be aware that horizontal step groups may cause layout issues, such as overflow, on small viewports or when the page is zoomed in." @coraliefeil I have added this hint to solve this issue: 

"Problem: Horizontal step groups are not responsive, which causes overflow on small viewports or when the page is zoomed in.
Solution: its the users responisbility => comment in the description should be added (tba by @coraliefeil)"

Is it okay? I have added it inside the orientation section as a hint. Please validate so dev can apply.
